edx/analytics/tasks/course_blocks.py

```diff
@@ -0,0 +1,501 @@
+"""
+Store course block details sourced from the Course Blocks API into a hive table.
+
+See LoadAllCourseBlocksWorkflow for details.
+"""
+
+import logging
+import json
+import datetime
+import luigi
+
+from edx.analytics.tasks.mapreduce import MapReduceJobTask, MapReduceJobTaskMixin
+from edx.analytics.tasks.url import get_target_from_url, url_path_join
+from edx.analytics.tasks.util.hive import (
+    WarehouseMixin, BareHiveTableTask, HivePartitionTask,
+)
+from edx.analytics.tasks.util.overwrite import OverwriteOutputMixin
+from edx.analytics.tasks.util.record import Record, BooleanField, StringField, IntegerField, DEFAULT_NULL_VALUE
+from edx.analytics.tasks.decorators import workflow_entry_point
+from edx.analytics.tasks.course_list import TimestampPartitionMixin, CourseListPartitionTask
+
+
+log = logging.getLogger(__name__)
+
+
+class CourseBlockRecord(Record):
+    """
+    Represents a course block as fetched from the edX Course Blocks REST API, augmented with details about its
+    position in the course.
+    """
+    block_id = StringField(length=255, nullable=False, description='Block identifier.')
+    course_id = StringField(length=255, nullable=False, description='Identifier for the course containing the block.')
+    block_type = StringField(length=255, nullable=False, description='Block type.')
+    display_name = StringField(length=255, nullable=False, description='User-facing title of the block.')
+    is_root = BooleanField(default=False, nullable=False,
+                           description='True if the block is the course\'s root node.')
+    has_multiple_parents = BooleanField(default=False, nullable=False,
+                                        description='True if the block has more than one parent, making the course '
+                                                    'a directed acyclic graph.  If True, then parent_block_id and '
+                                                    'course_path will be null.')
+
+    parent_block_id = StringField(length=255, nullable=True,
+                                  description='Block identifier for the block\'s parent. If null, and '
+                                              'is_root and has_multiple_parents are False, then this block has been '
+                                              'removed from the course.')
```

Comment:
Good point.  I added `is_orphan`, and so that made me want to rename `has_multiple_parents` to `is_dac`, which looks a little more cryptic, but makes the boolean field names nice and consistent.

See [99ee83b](https://github.com/open-craft/edx-analytics-pipeline/pull/4/commits/99ee83b33f7b100b3bfb5aab83a67d606cad27d9).